packages/context/src/binding.ts

```diff
@@ -94,6 +120,52 @@ export class Binding {
    * ```
    */
   getValue(ctx: Context): BoundValue | Promise<BoundValue> {
+    // First check cached value for non-transient
+    if (this._cache !== undefined) {
+      if (this._scope === BindingScope.SINGLETON) {
+        return this._cache;
+      } else if (this._scope === BindingScope.CONTEXT) {
+        if (ctx.contains(this._key)) {
+          return this._cache;
+        }
+      }
+    }
+    if (this._getValue) {
```

Comment:
This looks like a wall of code that's difficult to understand quickly. Can you split it into multiple smaller pieces (methods) please? We should also look for opportunities for code re-use. For example, I think the following block of code is used twice and can be extracted into a single shared function:

```ts
           if (ctx.contains(this._key)) {
              // The ctx owns the binding
              this._cache = val;
            } else {
              // Create a binding of the cached value for the current context
              ctx.bind(this._key).to(val);
            }
```